File: 83.删除排序链表中的重复元素.py
#
# @lc app=leetcode.cn id=83 lang=python
#
# [83] 删除排序链表中的重复元素
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head: return head
        prev = head
        p = head.next
        while p:
            if p.val == prev.val:
                prev.next = p.next
            else:
                prev = p
            p = p.next
        return head
# @lc code=end

File: 83.删除排序链表中的重复元素.c
/*
 * @lc app=leetcode.cn id=83 lang=c
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* deleteDuplicates(struct ListNode* head){
    if (head)
    {
        struct ListNode * prev = head;
        struct ListNode * p = head->next;
        while (p)
        {
            if (p->val == prev->val)
            {
                prev->next = p->next;
            }
            else
            {
                prev = p;
            }
            p = p->next;
        }
    }
    return head;
}


// @lc code=end

